lib/puppet/configurer.rb

```diff
@@ -126,6 +126,82 @@ def convert_catalog(result, duration, facts, options = {})
     catalog
   end
 
+  def check_fact_name_length(name)
+    unless Puppet[:max_factnamelength] == -1 
+      if Puppet[:max_factnamelength] == 0
+        Puppet[:max_factnamelength] = 100
+        if name.length > Puppet[:max_factnamelength]
+          Puppet.warning _("Fact %{name} exceedes the lenght limit: %{limit}") % { name: name, limit: Puppet[:max_factnamelength] }
+        end
+      else
+        if name.length > Puppet[:max_factnamelength]
+          Puppet.warning _("Fact %{name} exceedes the lenght limit: %{limit}") % { name: name, limit: Puppet[:max_factnamelength] }
+        end 
+      end
+    end
+  end
```

Comment:
You could simplify the nested `ifs` by using early returns:
```ruby
# unless Puppet[:max_factnamelength] == -1
return if Puppet[:max_factnamelength]
```

Since you need the value in `Puppet[:max_factnamelength]` you can store it in a local variable and adjust the value based on your needs:
```ruby
max_length = Puppet[:max_factnamelength]
return if max_length == -1 # early return, do nothing

max_length = 100 if max_length == 0 # set de default
...
# use whatever was set in `Puppet[:max_factnamelength]`

```

Finally, the error message can be extracted in a separate method and the whole thing can be rewritten as:
```ruby
def check_fact_name_length(name)
  max_length = Puppet[:max_factnamelength]
  return if max_length == -1

  max_length = 100 if max_length == 0
  warn_name_length(name,  max_length) if name.length > max_length
end

def warn_name_length(name, max_length)
    Puppet.warning _("Fact %{name} exceedes the lenght limit: %{limit}") % { name: name, limit: max_length }
end
```


Same mechanism can be applied for `check_fact_values_length` and `check_size_limits` methods